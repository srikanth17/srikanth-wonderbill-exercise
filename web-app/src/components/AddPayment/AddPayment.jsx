import React from 'react';
import styled from 'styled-components';
import { useDispatch } from 'react-redux';
import moment from 'moment';
import Container from '../Container';
import Header from '../Header/Header';
import PaymentForm from '../PaymentForm/PaymentForm';
import { postPayment } from '../../state/payments';
import theme from '../../theme';

const CustomContainer = styled(Container)`
  max-width: 350px;
`;

const Heading = styled.h2`
  text-align: center;
  color: ${theme.palette.text.primary};
  font-weight: normal;
  font-size: ${theme.spacing(6)};
`;

const SubHeading = styled.p`
  text-align: center;
  color: ${theme.palette.text.secondary};
`;

const AddPayment = ({ history }) => {
  const dispatch = useDispatch();
  const handleSubmit = payment => {
    const data = {
      ...payment,
      startDate: moment(payment.startDate).format(),
    };
    dispatch(postPayment(data));
    history.push('/');
  };

  return (
    <div>
      <Header heading="Add A Bill" />
      <CustomContainer>
        <Heading>Enter your details</Heading>
        <SubHeading>
          Keep track of your household spending by adding our bills
        </SubHeading>
        <PaymentForm onSubmit={handleSubmit} />
      </CustomContainer>
    </div>
  );
};

export default AddPayment;
