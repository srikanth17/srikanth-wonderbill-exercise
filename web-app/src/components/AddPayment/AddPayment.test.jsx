import React from 'react';
import { shallow } from 'enzyme';
import { useSelector, useDispatch } from 'react-redux';
import AddPayment from './AddPayment';

const payments = [
  {
    name: 'Payment1',
    amount: '56',
    startDate: '2021-01-13T12:00:00+00:00',
    frequency: 'Monthly',
  },
];

const mockDispatch = jest.fn();
jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
  useDispatch: () => mockDispatch,
}));

let history, wrapper;

beforeEach(() => {
  history = { push: jest.fn() };
  wrapper = shallow(<AddPayment history={history} />);
});

test('should render AddPayment correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should handle form submit', () => {
  wrapper.find('PaymentForm').prop('onSubmit')(payments[0]);
  expect(history.push).toHaveBeenLastCalledWith('/');
  expect(mockDispatch).toHaveBeenLastCalledWith(expect.any(Function));
});
