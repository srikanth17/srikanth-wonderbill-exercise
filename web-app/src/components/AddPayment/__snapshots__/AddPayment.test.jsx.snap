// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render AddPayment correctly 1`] = `ShallowWrapper {}`;
